ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/NetEnode.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.jsonrpc.internal.methods;
+
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcError;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcErrorResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.p2p.P2pDisabledException;
+import tech.pegasys.pantheon.ethereum.p2p.api.P2PNetwork;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class NetEnode implements JsonRpcMethod {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final P2PNetwork p2pNetwork;
+
+  public NetEnode(final P2PNetwork p2pNetwork) {
+    this.p2pNetwork = p2pNetwork;
+  }
+
+  @Override
+  public String getName() {
+    return "net_enode";
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequest req) {
+    try {
+      final StringBuilder response = new StringBuilder();
+      p2pNetwork
+          .getAdvertisedPeer()
+          .ifPresent(
+              advertisedPeer -> {
+                response.append(advertisedPeer.getEnodeURI());
+              });
+      return new JsonRpcSuccessResponse(req.getId(), response.toString());
```

Comment:
I'll discuss with the team what we should do when peer discovery is disabled (and getAdvertisedPeer() returns empty). I'm not sure if returning an empty string is the right approach.
Anyway, this shouldn't block us from merging this PR after addressing the other bits.

Reply:
Take a look at `EthHashrate`. When someone requests the hash rate that the node is mining. When we aren't mining, we return an error json rpc response with a message explaining why. We should do something similar here.

Reply:
I have refactored to check isP2pEnabled and return an appropriate  error response. 
Still have a question if P2P is enabled but the enode cannot be found.
For now I am simply returning P2P disabled there as well, but I left it separate so that can easily be changed if needed.